src/content/en/updates/2017/03/custom-scrollbar.md

```diff
@@ -0,0 +1,380 @@
+I’m not going into detail what they are and why they work, but you can think of
+them like 3D coordinates with an additional, fourth coordinate called _w_. This
+coordinate should be 1 except if you want to have perspective distortion. So all
+points are 4-dimensional vectors [x, y, z, w=1] and consequently matrices need
+to be 4x4 as well.
+
+One occasion where you can see that CSS uses homogeneous coordinates under the
+hood is when you define your own 4x4 matrices in a transform property using the
+`matrix3d()` function. `matrix3d` takes 16 arguments (because the matrix is
+4x4), specifying one column after the other. So we can use this function to
+manually specify rotations, translations etc. But what it also allows us to do
+is mess around with that _w_ coordinate!
+
+Before we can make use of `matrix3d()`, we need a 3D context – because without a
+3D context there wouldn’t be any perspective distortion and no need for
+homogeneous coordinates. To create a 3D context we need a container with a
+`perspective` and some elements inside that we can transform in the newly
+created 3D space. For
+[example](https://googlechrome.github.io/ui-element-samples/custom-scrollbar/step-1.html):
+
+<img src="/web/updates/images/2017/03/custom-scrollbar/perspective.png"
+  alt="A piece of CSS code that distorts a div using CSS’
+    perspective attribute.">
+
+The elements inside a perspective container are processed as follows:
+
+* Turn each corner (vertex) of an element into homogenous coordinates
+  `[x,y,z,w]` (relative to the perspective container)
+* Apply all of the element’s transforms as matrices from _right to left_
+* If the perspective element is scrollable, apply a scroll matrix
+* Apply the perspective matrix
+
+The scroll matrix is a translation along the y axis. If we _scroll down_ by
+400px, all elements need to be _moved up_ by 400px. The perspective matrix is a
+matrix that “pulls” points closer to the vanishing point the further back in 3D
+space they are. This achieves both effects of making things appear smaller when
+they are farther back and also makes them “move slower” when being translated.
+So if an element is pushed back, a translation of 400px will cause the element
+to only move 300px on the screen.
+
+If you want to know _all_ the details, you should read the
+[spec](https://drafts.csswg.org/css-transforms-2/#transform-rendering) on CSS’
+transform rendering model, but for the sake of this article I simplified the
+algorithm above.
+
+Our box is inside a perspective container with value p for the `perspective`
+attribute, and let’s assume the container is scrollable and is scrolled down by
+∇ pixels.
+
+<img src="/web/updates/images/2017/03/custom-scrollbar/matrixmath01.svg"
+  alt="Perspective matrix times scroll matrix times element transform matrix
+  equals four by four identity matrix with minus one over p in the fourth row
+  third column time four by four identity matrix with minus nabla in the second
```

Comment:
"nabla, is an operator used in mathematics, in particular, in vector calculus, as a vector differential operator," That argues for a brief explanation on first use rather than eliminating. Also, you might run start-appengine.sh and verify that it actually renders.